verify-push-backend/assets/index.html

```diff
@@ -255,30 +311,41 @@ <h3>Factors</h3>
         identity: $("#identity").val(),
       });
 
-      digestMessage($("#identity").val())
-        .then((identity) => {
-          const data = new URLSearchParams();
-          data.append("identity", identity);
-          data.append("message", $("#message").val());
-          data.append("factor", $("#factor").val());
+      const identity = $("#identity").val()
+      const data = new URLSearchParams();
+      data.append("identity", identity);
+      data.append("message", $("#message").val());
+      data.append("factor", $("#factor").val());
 
-          var details = $(".detail")
-            .toArray()
-            .map((detail) => detail.value)
-            .reduce(function (result, value, index, array) {
-              if (index % 2 === 0) result.push(array.slice(index, index + 2));
-              return result;
-            }, []);
+      var hiddenDetails = $(".hidden-detail")
+        .toArray()
+        .map((hiddenDetail) => hiddenDetail.value)
+        .reduce(function (result, value, index, array) {
+          if (index % 2 === 0) {
+            let key = array[index];
+            if (key !== "") result[key] = array[index + 1];
+          }
+          return result;
+        }, {});
+          
+      data.append("hiddenDetails", JSON.stringify(hiddenDetails));
```

Comment:
why are we using `JSON.stringify` here and appending tuples for `details` below? (Apologies it's been a while since I looked at this code)

Reply:
When creating a challenge, hiddenDetails is expected as a stringified JSON: https://www.twilio.com/docs/verify/api/challenge#create-a-challenge-resource